packages/react-charting/src/components/HorizontalBarChart/HorizontalBarChart.base.tsx

```diff
@@ -288,19 +295,47 @@ export class HorizontalBarChartBase extends React.Component<IHorizontalBarChartP
     );
   }
 
+  // Adjusts the margin percentage, based on the width of the svg (inversly propotional)
+  private _adjustBarSpacing(): void {
+    const svgWidth = this.barChartSvgRef.current?.getBoundingClientRect().width;
+    if (svgWidth) {
+      // [0,1] Controls the spacing between the bars
+      const BAR_SPACING_FACTOR = 1;
+      // The barspacing is provided in percentage terms, so changes inversely to the width of the svg
+      const currentBarSpacing = clamp((200 / svgWidth) * BAR_SPACING_FACTOR, 1, 0.2);
+      this.setState({ barSpacing: currentBarSpacing });
+    }
+  }
+
+  /**
+   * This functions returns an array of <rect> elements, which form the bars
+   * For each bar an x value, and a width needs to be specified
+   * The computations are done based on percentages
+   * Extra margin is also provided, in the x value to provide some spacing in between the bars
+   */
+
   private _createBars(data: IChartProps, palette: IPalette): JSX.Element[] {
+    const noOfBars =
+      data.chartData?.reduce(
+        (count: number, point: IChartDataPoint) => (count += (point.horizontalBarChartdata?.x || 0) > 0 ? 1 : 0),
+        0,
+      ) || 1;
+    const totalSpacing = this.state.barSpacing * (noOfBars - 1);
     const defaultPalette: string[] = [palette.blueLight, palette.blue, palette.blueMid, palette.red, palette.black];
     // calculating starting point of each bar and it's range
     const startingPoint: number[] = [];
-    const total = data.chartData!.reduce(
+    let total = data.chartData!.reduce(
       (acc: number, point: IChartDataPoint) =>
         acc + (point.horizontalBarChartdata!.x ? point.horizontalBarChartdata!.x : 0),
       0,
     );
+
+    total += (totalSpacing * total) / (100 - totalSpacing);
```

Comment:
I did not get this part. Why is new total a function of total spacing. And at this point total is before scaling it to 100.

Reply:
So the math I used for this part works out as

- the total is currently just the raw total of the elements (before it's being scaled to 100)
- the totalMargin calculations are done in percentages (for better clarity)
- Now the current total + (totalMargin%) will give us the new raw total
- But totalMargin% is the percentage of the final raw total, and not the current raw total
- So a mathematical equation is formed and needs to be solved to get precise calculations for margins
- Attaching the diagram and the equations I used here, lmk if there's something we need to fix or discuss in more detail.


![image](https://github.com/microsoft/fluentui/assets/70366079/f00735ab-d8da-49ab-a993-55e983d6cf12)

This ensures that once the raw total is scaled to 100, the width of bars all scale down proportionally, giving the required space for margins